### PR TITLE
Ej nc news task7a

### PR DESCRIPTION
**Pull request by EdJackson98**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -95,4 +95,46 @@ describe('/api/articles', () => {
...
+            body: 'Test comment',
+        };
+        return request(app)
+            .post('/api/articles/1/comments')
+            .send(testComment)
+            .expect(201)
+            .then((response) => {
+                expect(response.body.comment.author).toBe(testComment.username);
+                expect(response.body.comment.body).toBe(testComment.body);
+            });
```
Comment (dappernerddesigns) [inline __tests__/app.test.js:169]: I'd also add an expect here to check that the correct article id has been returned on the object.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -95,4 +95,46 @@ describe('/api/articles', () => {
...
+        };
+        return request(app)
+            .post('/api/articles/1/comments')
+            .send(incompleteComment)
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe('Bad Request: Missing field');
+            });
+    });
+});
```
Comment (dappernerddesigns) [inline __tests__/app.test.js]: We do need a 400 here as well for /api/articles/bananas or something like that. I know it's handled on your get, we're now on a post :)

File: controllers/articles.controller.js [hunk 3]
```diff
@@ -19,3 +19,29 @@ exports.getAllArticles = (req, res, next) => {
     .catch(next)
 };
 
+// exports.postCommentByArticle = (req, res, next) => {
```
Comment (dappernerddesigns) [inline controllers/articles.controller.js]: Make sure you get rid of commented out code

File: controllers/articles.controller.js [hunk 4]
```diff
@@ -19,3 +19,29 @@ exports.getAllArticles = (req, res, next) => {
...
+//         .then((comment) => {
+//             res.status(201).send({ comment });
+//         })
+//         .catch(next);
+// };
+
+exports.postCommentByArticle = (req, res, next) => {
+    const { article_id } = req.params;
+    const { username, body } = req.body;
+    if (!username || !body) {
```
Comment (dappernerddesigns) [inline controllers/articles.controller.js]: If you look at the seed file, you'll see there is a not null constraint on both the username and body. This means if they're missing, psql will kick out an error code. You shouldn't need a custom error handler here if you've set up your psql error handling functions.

File: controllers/articles.controller.js [hunk 5]
```diff
@@ -19,3 +19,29 @@ exports.getAllArticles = (req, res, next) => {
...
+        return res.status(400).send({ msg: 'Bad Request: Missing field' });
+    }
+    checkIDExists(article_id)
```
Comment (dappernerddesigns) [inline controllers/articles.controller.js:25]: Could this be handled as a Promise.all?

**Review by dappernerddesigns — CHANGES_REQUESTED** [review]
A really good effort and very close. Just a few bits to tidy up if you have a look at the comments below.

**Review by EddTheDuck — APPROVED** [review]
nice work :)

**Pull request merged**